perfkitbenchmarker/providers/gcp/gcp_dataproc.py

```diff
@@ -75,35 +74,57 @@ def _Create(self):
                              self.cluster_id)
     if self.project is not None:
       cmd.flags['project'] = self.project
-    cmd.flags['num-workers'] = self.num_workers
-    if self.machine_type:
-      cmd.flags['worker-machine-type'] = self.machine_type
-      cmd.flags['master-machine-type'] = self.machine_type
+    cmd.flags['num-workers'] = self.spec.worker_group.vm_count
+
+    for group_type, group_spec in [
+        ('worker', self.spec.worker_group),
+        ('master', self.spec.master_group)]:
+      flag_name = group_type + '-machine-type'
+      cmd.flags[flag_name] = group_spec.vm_spec.machine_type
+
+      if group_spec.vm_spec.num_local_ssds:
+        ssd_flag = 'num-{0}-local-ssds'.format(group_type)
+        cmd.flags[ssd_flag] = group_spec.vm_spec.num_local_ssds
+
+      if group_spec.vm_spec.boot_disk_size:
+        disk_flag = group_type + '-boot-disk-size'
+        cmd.flags[disk_flag] = group_spec.vm_spec.boot_disk_size
+
     cmd.Issue()
 
   def _Delete(self):
     """Deletes the cluster."""
     cmd = util.GcloudCommand(self, 'dataproc', 'clusters', 'delete',
                              self.cluster_id)
+    cmd.flags['zone'] = []
```

Comment:
It makes it so that no `--zone ....` is added to the command.  A `--zone` breaks some dataproc commands.